matrix_benchmarking/common.py

```diff
@@ -40,18 +82,36 @@ def settings_to_key(settings):
         return "|".join(f"{k}={settings[k]}" for k in sorted(settings) if k != "stats")
 
     @staticmethod
-    def all_records(settings, setting_lists):
+    def all_records(settings, setting_lists, include_local=True, include_lts=False) -> Iterator[MatrixEntry]:
         for settings_values in sorted(itertools.product(*setting_lists)):
             settings.update(dict(settings_values))
             key = Matrix.settings_to_key(settings)
+            logging.debug(f"Key: {key}")
             try:
-                yield Matrix.processed_map[key]
+                entry = Matrix.processed_map[key]
             except KeyError:
                 continue # missing experiment
+            if (entry.is_lts and include_lts) or (not entry.is_lts and include_local):
+                yield entry
 
     @staticmethod
     def get_record(settings):
         key = Matrix.settings_to_key(settings)
 
         try: return Matrix.processed_map[key]
         except KeyError: return None
+
+    @staticmethod
+    def count_records(settings, setting_lists, include_local=True, include_lts=False):
+        count = 0
+        for settings_values in sorted(itertools.product(*setting_lists)):
+            settings.update(dict(settings_values))
+            key = Matrix.settings_to_key(settings)
+            try:
+                entry = Matrix.processed_map[key]
+            except KeyError:
+                continue # missing experiment
+            if (entry.is_lts and include_lts) or (not entry.is_lts and include_local):
+                count += 1
+        return count
```

Comment:
I think this is equivalent to:
```suggestion
        return len(self.all_records(settings, setting_lists, include_local, include_lts))
```

could you also add this:
```
    @staticmethod
    def has_records(settings, setting_lists, include_local=True, include_lts=False):
         for _ in  self.all_records(settings, setting_lists, include_local, include_lts):
              return True
         return False
```
(this one avoid unnecessary traversal of the matrix, which can be useful where there are hundreds of records)